src/main/java/io/pravega/connectors/flink/FlinkPravegaWriter.java

```diff
@@ -174,37 +550,47 @@ public void invoke(T event) throws Exception {
                     executorService
             );
         }
-    }
 
-    @Override
-    public void snapshotState(FunctionSnapshotContext context) throws Exception {
-        if (this.writerMode == PravegaWriterMode.ATLEAST_ONCE) {
+        @Override
+        public void close() throws Exception {
+            flushAndVerify();
+            pravegaWriter.close();
+        }
+
+        @Override
+        public List<UUID> snapshotState(long checkpointId, long checkpointTime) throws Exception {
```

Comment:
Given that this writer isn't used when snapshots are enabled, is this method dead code?